test/functional/wallet_backwards_compatibility.py

```diff
@@ -306,86 +307,143 @@ def run_test(self):
-
-            # Restore the wallet to master
-            load_res = node_master.restorewallet(wallet_name, backup_path)
-
-            # Make sure this wallet opens with only the migration warning. See https://github.com/bitcoin/bitcoin/pull/19054
-            if not self.options.descriptors:
-                # Legacy wallets will have only a deprecation warning
-                assert_equal(load_res["warnings"], ["Wallet loaded successfully. The legacy wallet type is being deprecated and support for creating and opening legacy wallets will be removed in the future. Legacy wallets can be migrated to a descriptor wallet with migratewallet."])
-            else:
-                assert "warnings" not in load_res
+                    if self.major_version_at_least(node, 21):
+                        node.rpc.createwallet(wallet_name=wallet_name, descriptors=self.options.descriptors)
+                    else:
+                        node.rpc.createwallet(wallet_name=wallet_name)
+                    wallet_prev = node.get_wallet_rpc(wallet_name)
+                    address = wallet_prev.getnewaddress('', "bech32")
+                    addr_info = wallet_prev.getaddressinfo(address)
+
+                if encrypt:
+                    # Mock time forward so that the new active descriptors have a distinct timestamp
+                    node.setmocktime(int(time.time()) + 60)
+                    wallet_prev.encryptwallet("pass")
+
+                if self.options.descriptors and self.major_version_at_least(node, 23):
+                    # Retrieve the xprv that is expected for the wallet hd key upgrade
+                    # This depends on listdescriptors(True) which requires at least 23.0
+                    if encrypt:
+                        wallet_prev.walletpassphrase("pass", 100000)
+                    descs = wallet_prev.listdescriptors(True)["descriptors"]
+                    for desc in descs:
+                        if desc["active"] and desc["desc"].startswith("wpkh("):
+                            expected_xprv = desc["desc"][5:116]
+                            break
+                    # Import a few new descriptors as active, but not enough to be considered the active hd key
+                    wallet_prev.importdescriptors([
+                        {
+                            "desc": descsum_create("wpkh(tprv8ZgxMBicQKsPeNLUGrbv3b7qhUk1LQJZAGMuk9gVuKh9sd4BWGp1eMsehUni6qGb8bjkdwBxCbgNGdh2bYGACK5C5dRTaif9KBKGVnSezxV/0/*"),
+                            "timestamp": int(time.time()) + 70,
+                            "active": True,
+                        },
+                        {
+                            "desc": descsum_create("wpkh(tprv8ZgxMBicQKsPeNLUGrbv3b7qhUk1LQJZAGMuk9gVuKh9sd4BWGp1eMsehUni6qGb8bjkdwBxCbgNGdh2bYGACK5C5dRTaif9KBKGVnSezxV/1/*"),
+                            "timestamp": int(time.time()) + 70,
+                            "active": True,
+                            "internal": False,
```

Comment:
In 4ad97b519e7:
As the default value for "internal" is false, I infer you wanted `internal=true`?.

Same for the other one that is below.

Reply:
Ah, good catch. Fixed